examples/2d/jeans_instability.jl

```diff
@@ -0,0 +1,100 @@
+
+
+###############################################################################
+# semidiscretization of the hyperbolic diffusion equations
+resid_tol = 1.0e-4
+equations_gravity = HyperbolicDiffusionEquations2D(resid_tol)
+
+solver_gravity = DGSEM(polydeg, flux_lax_friedrichs)
+
+semi_gravity = SemidiscretizationHyperbolic(mesh, equations_gravity, initial_conditions, solver_gravity,
+                                            source_terms=source_terms_harmonic)
+
+
+###############################################################################
+# combining both semidiscretizations for Euler + self-gravity
+parameters = ParametersEulerGravity(background_density=1.5e7, # aka rho0
+                                    gravitational_constant=6.674e-8, # aka G
+                                    cfl=2.4,
+                                    n_iterations_max=1000,
+                                    timestep_gravity=timestep_gravity_erk52_3Sstar!)
+
+semi = SemidiscretizationEulerGravity(semi_euler, semi_gravity, parameters)
+
+
+###############################################################################
+# ODE solvers, callbacks etc.
+tspan = (0.0, 5.0)
+ode = semidiscretize(semi, tspan);
+
+summary_callback = SummaryCallback()
+
+analysis_interval = 100
+alive_callback = AliveCallback(analysis_interval=analysis_interval)
+
+Trixi.pretty_form_repl(::Val{:energy_potential}) = "∑e_potential"
+Trixi.pretty_form_file(::Val{:energy_potential}) = "e_potential"
+
+function Trixi.analyze(::Val{:energy_potential}, du, u_euler, t, semi::SemidiscretizationEulerGravity)
+
+  u_gravity = Trixi.wrap_array(semi.cache.u, semi.semi_gravity)
+
+  mesh, equations_euler, dg, cache = Trixi.mesh_equations_solver_cache(semi.semi_euler)
+  _, equations_gravity, _, _ = Trixi.mesh_equations_solver_cache(semi.semi_gravity)
+
+  e_potential = Trixi.integrate(mesh, equations_euler, dg, cache, u_euler, equations_gravity, u_gravity) do u, i, j, element, equations_euler, dg, equations_gravity, u_gravity
+    u_euler_local   = Trixi.get_node_vars(u_euler,   equations_euler,   dg, i, j, element)
+    u_gravity_local = Trixi.get_node_vars(u_gravity, equations_gravity, dg, i, j, element)
+    # OBS! subtraction is specific to Jeans instability test where rho0 = 1.5e7
+    return (u_euler_local[1] - 1.5e7) * u_gravity_local[1]
+  end
+  return e_potential
+end
+
+analysis_callback = AnalysisCallback(semi_euler, analysis_interval=analysis_interval,
+                                     save_analysis=true,
+                                     extra_analysis_integrals=(entropy, energy_total, energy_kinetic, energy_internal, Val(:energy_potential)))
+
+save_solution = SaveSolutionCallback(solution_interval=10,
+                                     save_initial_solution=true,
+                                     save_final_solution=true,
+                                     solution_variables=:primitive)
```

Comment:
Is there a particular reason why it's not `save_solution_callback`?

And why is `solution_variables` a symbol and not a function? Or is this just one of the things on the TODO pile for us?

Reply:
`save_solution_callback` just was a bit too long and broke the symmetry of the names used for other solvers, being composed of just two words... We can definitely call it `save_solution_callback` for (another kind of) consistency if you like.

`solution_variables` is a symbol right now because that's how it was handled in `save_solution` before :sweat_smile: It's on my radar to make this more generic, but I didn't want to write the code and decide on a final interface yet, see
```julia
# TODO: Taal, refactor, allow saving arbitrary functions of the conservative variables,
# make solution_variables a function instead of a Symbol
```
I think the interface would look similar to different analysis quantities: Define it as something that is called as `solution_variables(u, equations)` and define 
```julia
pretty_form_file(::typeof(solution_variables)) = #= tuple or something simiar of strings containing the variable names =#
```

Reply:
:+1: